tests/worker/test_workflow.py

```diff
@@ -7211,8 +7211,8 @@ async def test_workflow_deadlock_fill_up_slots(client: Client):
         client,
         DeadlockFillUpBlockWorkflow,
         DeadlockFillUpSimpleWorkflow,
-        # Start the worker with CPU count + 10 task slots
-        max_concurrent_workflow_tasks=cpu_count + 10,
+        # Start the worker with CPU count + 11 task slots
+        max_concurrent_workflow_tasks=cpu_count + 11,
```

Comment:
I'll admit I don't actually understand how starting `cpu_count + 5` workflows somehow eats up 100% of `cpu_count + 10` task slots.

However, it does, and before my change the pollers would (erroneously) make sure permits were attempted to be acquired evenly between sticky/nonsticky. Now that that's fixed, 100% of the permits can be eaten up by the deadlockers, meaning the non-deadlocking workflow never gets a chance to start.

This is, IMO, seemingly an natural consequence of deadlocking every single task slot. From what I understand, we all accepted that we could never free slots of deadlocked workflows until they actually became released.

So, seemingly this test change is acceptable, but please LMK if I'm wrong.

Reply:
Ah, yes, that is it. I forgot the default number of pollers here is 5.